neuralmonkey/decoders/sequence_labeler.py

```diff
@@ -139,3 +140,59 @@ def feed_dict(self, dataset: Dataset, train: bool = False) -> FeedDict:
             fd[self.train_weights] = paddings.T
 
         return fd
+
+
+class CRFLabeler(SequenceLabeler):
+    """CRF Classifier assing a label to each encoder's state."""
+
+    def __init__(self,
+                 name: str,
+                 encoder: Union[RecurrentEncoder, SentenceEncoder],
```

Comment:
Nemůže to být něco obecnějšího? `TemporalStateful`?

Reply:
No potrebujes objekt, u ktereho mas zarucene, ze ma nadefinovane logity (nebo obecneji vystupni matici ve formatu [batch_size, seq_len]). Myslim, ze TemporalStateful tuhle podminku nesplnuje.

Reply:
temporal stateful to je, má to (batch, time, whatever) matici a masky. Nicméně distribuce přes slovník (ať už normalizovaný nebo ne) bych stavama spíš nenazýval.

pořád si myslim že to nemá dědit od sequence labeleru vůbec;  ta CRF vrstva mi přijde jen jako jinej způsob převodu logitů (batch, time, vocabulary) na symboly (batch, time) a loss (), kde default je argmax a sequence_loss. Nebo se pletu?

Reply:
Jo, misto po jednotlivych symbolech (tj. bez ohledu na dekodovane sousedy) se tady ucis dekodovat celou sekvenci najednou.
Jeste je otazka, jestli by byl problem stavet CRF vrstvu nad normalizovanymi logity (tedy po softmaxu). Pokud by to slo, tak by se tam daly implementovat i ensembly.

Reply:
Zamyslel bych se jak to abstrahovat, ale před tim bych teda nechal tohle pracovat s čímkoliv (`Any`) co má za runtimu atribut `logits`. Plus udělat spešl syntactic sugar verzi, která si udělá svuj sequence labeler, podobně jako si ``encoders.SentenceEncoder`` dělá svou ``EmbeddedSequence``.